ledger-core/src/CMakeLists.txt

```diff
@@ -8,7 +8,7 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 # Automatically add API files to the library
 file(
     GLOB ledger-core-api-sources
-    "${CMAKE_SOURCE_DIR}/inc/core/api/*.cpp"
+    "${CMAKE_SOURCE_DIR}/inc/core/api/*"
```

Comment:
Hm. Why do you need every file now? I’m not sure what `.hpp` are for here.

Reply:
api headers were included in `HEADERS_FILES` before
The change is 2 part, accumulated with 
```patch
-file(GLOB_RECURSE HEADERS_FILES ../inc/*.{h,hpp})
+file(GLOB_RECURSE HEADERS_FILES ../inc/*.hpp)
+list(REMOVE_ITEM HEADERS_FILES ../inc/api/*.hpp)
```
below.

Combined it has the effect of having `ledger-core-api-sources` be both `cpp` and `hpp` files, whereas `HEADERS_FILES` holds only non generated headers. The api headers were always included in the `ledger-core-obj` target.

Now that there are both a `ledger-core-obj` and a `ledger-core-api-obj` target, it makes sense to split the headers depending on being `api/` or `not`